lib/vdsm/storage/lvm.py

```diff
@@ -408,6 +421,11 @@ def _getCachedFilter(self):
     def _addExtraCfg(self, cmd, devices=tuple(), use_lvmpolld=True):
         newcmd = [constants.EXT_LVM, cmd[0]]
 
+        if USE_DEVICES:
+            allowed_devices = _allowed_devices(devices)
+            if allowed_devices:
+                newcmd += ["--devices", allowed_devices]
+
```

Comment:
Adding this without removing the filter from --config does not make sense. It would be better to first change the filter code to support a config without filter, and then in this patch add either --devices or the filter, not both.

This is the general approach - make the change easy, make the easy change.